### PR TITLE
[13.0][MIG] account_chart_update: Migration to 13.0

### PR DESCRIPTION
**Pull request by ernestotejeda**
Main changes:
- Refactorize the diff_fields method in wizards/wizard_chart_update.py because pre-commit determined it was too complex
- Adapt the python code of the wizard to take into account the use of the new fields 'Repartition for Invoices' and 'Repartition for Credit Notes' in Taxes.
- Adapt tests

Cc @Tecnativa TT19682

### CODE REVIEW

File: account_chart_update/wizard/wizard_chart_update.py [hunk 1]
```diff
@@ -375,7 +360,13 @@ def _get_real_xml_name(self, template):
         (name, module) = external_id.split(".")
         return "%s.%d_%s" % (name, self.company_id.id, module)
 
-    @api.multi
+    @tools.ormcache("templates")
+    def find_taxes_by_templates(self, templates):
```
Comment (yajo) [inline account_chart_update/wizard/wizard_chart_update.py:364]: Is this method meant to be called from RPC?

- Yes: then decorate with the corresponding `@returns('self')`
- No: then prefix it with underline to make it private: `_find_taxes_by_templates`

File: account_chart_update/wizard/wizard_chart_update.py [hunk 2]
```diff
@@ -375,7 +360,13 @@ def _get_real_xml_name(self, template):
...
+        tax_ids = []
+        for tax in templates:
+            tax_ids.append(self.find_tax_by_templates(tax))
+        return self.env["account.tax"].browse(tax_ids)
+
     @tools.ormcache("templates")
     def find_tax_by_templates(self, templates):
```
Comment (yajo) [inline account_chart_update/wizard/wizard_chart_update.py:371]: Same question as above... Well, just take a look at the whole file, it's full of these things.

**Review by yajo — CHANGES_REQUESTED** [review]
Code looks good, just a detail that repeats a few times.

File: account_chart_update/wizard/wizard_chart_update.py (hunk 1 above)
Comment (pedrobaeza) [reply account_chart_update/wizard/wizard_chart_update.py:364]: I think in v13 `@api.returns` is deprecated, and not sure why do you want to put `self` as argument. Any way, you did these methods (for v9 or for v10), so you tell us.

Comment (yajo) [reply account_chart_update/wizard/wizard_chart_update.py:364]: It was 4 years ago, I have no idea :grin:

Comment (pedrobaeza) [reply account_chart_update/wizard/wizard_chart_update.py:364]: OK, then discarding the suggestions. If we ever need that, we add it.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**